### ISSUE TITLE
fmt/color.h missing format_to overload

### ISSUE THREAD

**Issue by Sarcasm** [fmtlib/fmt#1593]
Hello,

I was wondering if there was any reason for fmt/color.h to not provide a format_to() overload?
I'd like to print messages to stderr using a style similar to GCC/Clang:

1. subject in bold
2. diagnostics kind in bold and colored: e.g. red for errors
3. message in normal text

However, due to multiprocessing issues, I'd like to write the messages "atomically" to stderr so output does not get mixed up.

Right now I can do it that way:

```cpp
std::string subject = fmt::format(fmt::emphasis::bold, "{}:", p);
std::string error = fmt::format(fg(fmt::color::red) | fmt::emphasis::bold, "error:");
fmt::print(stderr, "{} {} {}\n", subject, error, strerror(errno));
```

But it bothers me to allocate so many `std::string`, a color-supporting `format_to()` would be an improvement.

**Comment by vitaut** [comment]
It was proposed before (#1161) but noone was interested enough in implementing it probably because there is little value in optimizing allocations when you are dealing with I/O.

**Comment by vitaut** [comment]
PRs are welcome =).

**Comment by Sarcasm** [comment]
I see, I missed that ticket when searching.
I saw that there was a `vformat_to`, so I thought it will be peanuts to add, but I realize now that the signature takes a memory_buffer, not an output iterator.

It would be nice for completeness to have it, but I agree that colors are mostly used when printing to a tty, in which case I doubt the extra strings matter much, at least they do not for my use case.

Closing, I will let a future brave soul do it.

Thank you
